### ISSUE TITLE
Ycmd bundled clangd can't find the standard library on macOS Catalina

### ISSUE THREAD

**Issue by bstaletic** [clangd/clangd#211]
Hi,

ever since Catalina came out ycmd has been seeing users whose clangd can't find things like `stdio.h` or `iostream`. The clangd that ycmd downloads is located [here](https://bintray.com/ycm-core/clangd/clangd/9.0.0#files) and is just a repackaged Sierra archive from homebrew, because https://releases.llvm.org took too long to upload the macOS archive. However, today I had a ycmd user try the clangd from the releases.llvm.org and it also failed to find the standard library. On the other hand, if a user installs `clangd` by running `brew install clangd` and points ycmd to use that clangd, it all works.

To recap:

- Sierra homebrew repackaged and hosted on bintray - doesn't find `stdio.h` or `iostream`.
- Replacing the one above with the one from releases.llvm.org - same result.
- `brew install llvm` - works.

If a user is using an extra conf, then the logic that ycmd uses to find the standard library also works as clangd gets the `-isystem`/`-iframework` flags listed explicitly.

The logs are completely useless, but for what it's worth: https://gist.github.com/vegerot/ce5e97814c1d49c7a53ffd418e45bf20

#178 seems possibly related, but I'm not sure.

**Comment by sam-mccall** [comment]
I *strongly* suspect this is a dupe of #178 

Currently I think clang's driver logic only finds stdlib on mac if it gets the right path to clang, and `clang` == /usr/bin/clang isn't it, we have to run `xcrun -f clang`.
(I've heard that this might be more of a clean-install issue than a 10.15 issue - apparently 10.13 and maybe earlier didn't create /usr/include, but upgrade installs preserved it if it didn't exist).

A colleague reproduced with clang at head on a clean macbook. brew install llvm probably works because it installs stdlib in a more unixy location that we're able to find.

https://reviews.llvm.org/D70863 is a candidate fix.

**Comment by rubendibattista** [comment]
Hello, don't know if it's a regression, but I'm experiencing the same problem on Catalina with clangd-10 (within YouCompleteMe). C++ headers are not found, while C headers are ok.

The difference from the description of the previous problem is that I'm on Macports, not `brew`. So things get installed in `/opt/local` and then it does not fix the problem if I install `clang-10` from Macports.

I'm available to test things if it's needed.
